presentation/src/main/java/com/mashup/gabbangzip/sharedalbum/presentation/ui/common/PicDatePickerDialog.kt

```diff
@@ -0,0 +1,95 @@
+package com.mashup.gabbangzip.sharedalbum.presentation.ui.common
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.DatePicker
+import androidx.compose.material3.DatePickerDefaults
+import androidx.compose.material3.DatePickerDialog
+import androidx.compose.material3.DisplayMode
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.SelectableDates
+import androidx.compose.material3.rememberDatePickerState
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.DialogProperties
+import com.mashup.gabbangzip.sharedalbum.presentation.R
+import com.mashup.gabbangzip.sharedalbum.presentation.theme.Cultured
+import com.mashup.gabbangzip.sharedalbum.presentation.theme.Gray0
+import com.mashup.gabbangzip.sharedalbum.presentation.theme.Gray80
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun PicDatePickerDialog(
+    modifier: Modifier = Modifier,
+    date: Long? = null,
+    onClickedDismiss: () -> Unit = {},
+    onClickedConfirm: (Long) -> Unit = {},
+) {
+    val datePickerState = rememberDatePickerState(
+        yearRange = 2020..2030,
+        initialDisplayMode = DisplayMode.Picker,
+        initialSelectedDateMillis = date ?: System.currentTimeMillis(),
+        selectableDates = object : SelectableDates {
+            override fun isSelectableDate(utcTimeMillis: Long): Boolean = true
+        },
```

Comment:
```suggestion
        selectableDates = DatePickerDefaults.AllDates
```

```
@ExperimentalMaterial3Api
@Stable
interface SelectableDates {

    /**
     * Returns true if the date item representing the [utcTimeMillis] should be enabled for
     * selection in the UI.
     */
    fun isSelectableDate(utcTimeMillis: Long) = true

    /**
     * Returns true if a given [year] should be enabled for selection in the UI. When a year is
     * defined as non selectable, all the dates in that year will also be non selectable.
     */
    fun isSelectableYear(year: Int) = true
}
```
SelectableDates 의 isSelectableDate 값이 true 인 것 같아서 selectableDates 를 아예 default 값으로 쓰거나, 명시하고 싶다면 DatePickerDefaults.AllDates 을 사용하는 것도 나쁘지 않아보이는데 SelectableDates 익명객체를 만들어준 이유가 특별히 없다면 이런 방법은 어떨까

Reply:
헉 아주 좋은 생각. 안그래도 저거 좀 불편했는데 넘 좋아

Reply:
9a27a23a54d48e25af2618fce2eb5ab237281540